lib/facade/MonitoringAspect.ts

```diff
@@ -84,8 +84,8 @@ export class MonitoringAspect implements IAspect {
     }
   }
 
-  private getMonitoringDetails<T>(
-    aspectOptions?: MonitoringAspectType<T>,
+  private getMonitoringDetails<T extends Record<string, any>>(
+    aspectOptions?: BaseMonitoringAspectType & { props?: T },
```

Comment:
You mentioned removing generics, was that specifically for generics in interfaces or do we need to need to remove *all* generics (ie the `T` type you have on the function)

Reply:
https://aws.github.io/jsii/user-guides/lib-author/typescript-restrictions/#typescript-mapped-types

It's specifically usage in what ends up being part of the constructs' APIs. Internal usage is still fine (as evidenced by the satisfied build without JSII complaining about it), since that's still just always run within a Node.js context even if the consumer isn't directly using it.